scripts/generate

```diff
@@ -191,7 +191,9 @@ def('substitutions',
                           K(el('span', 'arrow', htmlEncode('~>')))],
                          [contains(_, ap([I, htmlEncode], ['->'])),
                           K(el('span', 'arrow',
-                               el('span', 'arrow-hyphen', htmlEncode('-')) +
+                               //  Use NON-BREAKING HYPHEN so arrow does
+                               //  not break because of line wrapping
+                               el('span', 'arrow-hyphen', '&#x2011;') +
```

Comment:
@safareli obviously not my call, but in my opinion Unicode symbols look too small in monospaced fonts. I'd rather see them the size closer to e.g. [Hasklig's ligatures](https://github.com/i-tu/Hasklig#hasklig).

For reference, here's how `filter` signature would look with Unicode equivalents:

![`filter` signature with Unicode symbols](https://cloud.githubusercontent.com/assets/389387/23183549/865d1692-f88d-11e6-9331-6d85bbf96787.png)